tests/tracer/test_sampler.py

```diff
@@ -33,6 +33,19 @@ def dummy_tracer():
     return DummyTracer()
 
 
+@pytest.fixture
+def mock_rate_limiter(mocker, mock_effective_rate):
+    mocker.patch.object(RateLimiter, "is_allowed")
+    yield RateLimiter
+
+
+@pytest.fixture
+def mock_effective_rate(mocker):
```

Comment:
Is there a particular reason to mock rather than subclass and use the object normally? Just a tad worried that we're too far displaced from reality here with the mocks.

Reply:
yeah, that is a good question... I don't think so, other than we cannot pass in a rate limiter object to the sampler. we would need to do `sampler.limiter = TestRateLimiter()` which isn't a _big_ deal, but maybe a place where a `rate_limiter_cls` type argument would be helpful?

Reply:
Yeah, to me it seems like a fault in the API if we're having to mock it to assert the correct functionality?

Reply:
or I can just do `rate_limit=0` and assert the effective rate is `0`, which is "fine"

Reply:
we have other tests that effective rate and actual rate limiter work, so doesn't need to be done here too